### DISCUSSION TITLE
Credit card form example - validation not working?

### DISCUSSION THREAD

**Discussion by jasonlotz** [charmbracelet/bubbletea#1146] [Q&A] [answered]
I'm working with the credit card form example to play with the validation of text inputs.  Unless I'm doing something very wrong, the example's text input validation is not working at all.  I added logging to check that the validate functions are getting called and returning errors, but I do not see any errors on the TUI.  I have never seen an example of validation on textinputs in action, so I'm not 100% sure how it should look on the screen, but nothing happens at all.  Has anyone else noticed this or am I missing something?

**Comment by AdditionAddict** [comment, marked as answer]
yeah, example doesn't show errors in the tui

`huh` has ability to show errors via validate
https://pkg.go.dev/github.com/charmbracelet/huh#section-readme
```go
    // Gather some final details about the order.
    huh.NewGroup(
        huh.NewInput().
            Title("What’s your name?").
            Value(&name).
            // Validating fields is easy. The form will mark erroneous fields
            // and display error messages accordingly.
            Validate(func(str string) error {
                if str == "Frank" {
                    return errors.New("Sorry, we don’t serve customers named Frank.")
                }
                return nil
            }),
```

**Reply by jasonlotz** [reply]
Much appreciated, thank you.  I will spend some time looking at huh.

**Comment by meowgorithm** [comment]
Wow, the example in Bubble Tea is totally broken. Let's fix that one posthaste.

**Reply by jasonlotz** [reply]
#This would be great, thanks.  I'm new to Bubbletea so I wasn't 100% sure what the expected behavior to be.  I added some logging and it appears that the validate methods are getting called and returning errors, but the Update calls on each input are not doing anything with the error message.

```	
for i := range m.inputs {
	m.inputs[i], cmds[i] = m.inputs[i].Update(msg)
}
```

Being new to Bubbletea, I'm not 100% sure what the expected behavior should be, but this has been a good exercise to look deeper into the code.  Fixing this would be ideal as it's a nice clean example of using input validation.
